game-core/src/main/java/games/strategy/triplea/ui/MovePanel.java

```diff
@@ -1216,6 +1210,30 @@ private void updateUnitsThatCanMoveOnRoute(final Collection<Unit> units, final R
     unitsThatCanMoveOnRoute = new ArrayList<>(bestWithDependents);
   }
 
+  // Whether the two units are equivalent for the purposes of movement.
+  private boolean unitsAreEquivalent(final Unit u1, final Unit u2) {
+    final BigDecimal left1 = TripleAUnit.get(u1).getMovementLeft();
+    final BigDecimal left2 = TripleAUnit.get(u2).getMovementLeft();
+    return u1.isEquivalent(u2) && left1.equals(left2);
```

Comment:
Thanks for suggesting testing land transports.

Indeed, it doesn't handle that case well. But actually even without this change, it seems it's buggy - that's because the sort order of units is buggy - so if you select 3 infantry, 1 artillery and 2 tanks (with mech infantry tech), it's not guaranteed the game figures out that sending those units 2 territories away allows the 2 infantry and 2 tanks to go - because of the arbitrary sort order of infantry and artillery! This is with the current code without this change.

But with this change, it doesn't work at all - thanks for noticing!

Let me see how I can support that, I suspect we just need logic to determine how many units could be assisted and make sure they're sorted last.

Reply:
I have an initial version that seems to solve it, though I'd like to do more testing, including another unit test.

Reply:
Ok. Once you think this is ready let me know. If you can add some unit tests and a few screenshots to this PR that would be very helpful. I'd like to test this a bit myself manually with a few different maps as well.